### PR TITLE
Fixup multi-collator parachain transition to async backing

### PR DESCRIPTION
**Pull request by alexggh**
Fixing: 
```
Verification failed for block 0x07bbf1e04121d70a4bdb21cc055132b53ac2390fa95c4d05497fc91b1e8bf7f5 received from (12D3KooWJzLd8skcAgA24EcJey7aJAhYctfUxWGjSP5Usk9wbpPZ): "Header 0x07bbf1e04121d70a4bdb21cc055132b53ac2390fa95c4d05497fc91b1e8bf7f5 rejected: too far in the future"   
```

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5382094

File: substrate/bin/node-template/node/src/service.rs [hunk 1]
```diff
@@ -80,14 +80,12 @@ pub fn new_partial(config: &Configuration) -> Result<Service, ServiceError> {
...
 
-	let slot_duration = sc_consensus_aura::slot_duration(&*client)?;
-
 	let import_queue =
 		sc_consensus_aura::import_queue::<AuraPair, _, _, _, _, _>(ImportQueueParams {
 			block_import: grandpa_block_import.clone(),
 			justification_import: Some(Box::new(grandpa_block_import.clone())),
 			client: client.clone(),
-			create_inherent_data_providers: move |_, ()| async move {
+			create_inherent_data_providers: move |_, slot_duration| async move {
```
Comment (bkchr) [inline substrate/bin/node-template/node/src/service.rs:88]: The first parameter is the parent hash. So, just take this parent hash to get the slot duration here. No need to do all the other changes.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5384856

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5384874

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5384873

**Comment by s0me0ne-unkn0wn** [comment]
@bkchr looks legit now? Did some local tests, seems to work as expected.

**Review by s0me0ne-unkn0wn — APPROVED** [review]

File: prdoc/pr_3510.prdoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+title: "Fix multi-collator parachain transition to async backing"
+
+doc:
+  - audience: Node Dev
```
Comment (bkchr) [inline prdoc/pr_3510.prdoc:4]:
```suggestion
  - audience: Node Operator
```

**Review by bkchr — APPROVED** [review]
Yeah looks good! Ty @s0me0ne-unkn0wn!

**Pull request merged**